lib/Backend/LowerMDShared.cpp

```diff
@@ -1895,13 +1915,38 @@ void LowererMD::LegalizeSrc(IR::Instr *const instr, IR::Opnd *src, const uint fo
             return;
 
         case IR::OpndKindIntConst:
-            Assert(!instr->isInlineeEntryInstr);
-            if(forms & L_Imm32)
+            if(forms & L_Ptr)
             {
                 return;
             }
+#ifdef _M_X64
+            {
+                IR::IntConstOpnd * intOpnd = src->AsIntConstOpnd();
+                if ((forms & L_Imm32) && ((TySize[intOpnd->GetType()] != 8) ||
+                    (!instr->isInlineeEntryInstr && Math::FitsInDWord(intOpnd->GetValue()))))
+                {
+                    // the constant fits in 32-bit, no need to hoist
+                    return;
+                }
+                if (verify)
+                {
+                    AssertMsg(false, "Missing legalization");
+                    return;
+                }
+                // The actual value for inlinee entry instr isn't determined until encoder
+                // So it need to be hoisted conventionally.
+                if (!instr->isInlineeEntryInstr)
+                {
+                    Assert(forms & L_Reg);
+                    IR::IndirOpnd * indirOpnd = instr->m_func->GetTopFunc()->GetConstantAddressIndirOpnd(intOpnd->GetValue(), IR::AddrOpndKindConstantAddress, TyMachPtr, Js::OpCode::MOV);
```

Comment:
>intOpnd->GetValue() [](start = 105, length = 19)

This is still being treated as an address inside the API. and it looks like the API is creating AddrOpnd. Should you be creating a IntOpnd in this case ?